src/pynxtools/nomad/schema.py

```diff
@@ -137,6 +138,82 @@ class NexusActivityStep(ActivityStep):
     )
 
 
+class NexusIdentifier(EntityReference):
+    name = Quantity(
+        type=str,
+        description="The name of the nexus identifier field.",
+    )
+
+    def normalize(self, archive, logger):
+        from nomad.processing import Entry
+
+        def create_Entity(lab_id, archive, f_name, qunt_name):
+            entitySec = Entity()
+            entitySec.lab_id = lab_id
+            entitySec.name = qunt_name
+            entity = EntryArchive(
+                data=entitySec,
+                m_context=archive.m_context,
+                metadata=EntryMetadata(
+                    entry_type="identifier", domain="nexus", readonly=True
+                ),  # upload_id=archive.m_context.upload_id,
+            )
+            with archive.m_context.raw_file(f_name, "w") as f_obj:
+                json.dump(entity.m_to_dict(with_meta=True), f_obj)
+            archive.m_context.process_updated_raw_file(f_name)
+
+        def get_entry_reference(archive, f_name):
+            """Returns a reference to data from entry."""
+            from nomad.utils import hash
+
+            upload_id = archive.metadata.upload_id
+            entry_id = hash(upload_id, f_name)
+            return f"../uploads/{upload_id}/archive/{entry_id}#data"
+            # return f"/entries/{entry_id}/archive#/data"
+
+        f_name = f"{self.name}_{self.lab_id}.archive.json"
+        # Chekc if the entry already exists
+        entry = Entry.objects(
+            upload_id=archive.metadata.upload_id, mainfile=f_name
+        ).first()
+        if not entry:
+            create_Entity(self.lab_id, archive, f_name, self.name)
+        self.reference = get_entry_reference(archive, f_name)
+        super().normalize(archive, logger)
+
+
+class NexusReferences(ArchiveSection):
+    NXidentifiers = SubSection(
+        section_def=NexusIdentifier,
+        repeats=True,
+        description="These are the NOMAD references correspond to NeXus identifierNAME fields.",
+    )
+
+    def normalize(self, archive, logger):
+        # Consider multiple identifiers exists in the same group/section
+        identifiers = [
```

Comment:
One could get all identifiers by simply getting `identifierNAME` which is a dict of them.

Reply:
I am not sure section obeject has a dict for variadic name like `identifierNAME__field`,
```debug
>> self.__dict__["identifierNAME__field"]
Traceback (most recent call last):
  File "<string>", line 1, in <module>
KeyError: 'identifierNAME__field'

>>getattr(self, "identifierNAME__field")
```debug
or not with 
```
>> self.__dict__.get("identifierNAME")

>> getattr(self, "identifierNAME")

```